practice/tasks/IDisposable/IDisposableImplementation/IDisposableImplementation/MemoryStreamLogger.cs

```diff
@@ -1,22 +1,52 @@
-﻿using System.IO;
+﻿using System;
+using System.IO;
 
 namespace NetMentoring
 {
-    public class MemoryStreamLogger
+    public class MemoryStreamLogger : IDisposable
     {
-        private FileStream memoryStream;
-        private StreamWriter streamWriter;
+        private StreamWriter _streamWriter;
+        private StreamReader _streamReader;
+        private const string PathToFile = @"Files\log.txt";
+        private bool _disposed;
 
-        public MemoryStreamLogger()
+        public void ReadFile()
         {
-            memoryStream = new FileStream(@"\log.txt", FileMode.OpenOrCreate);
-            streamWriter = new StreamWriter(memoryStream);
+            _streamReader = new StreamReader(PathToFile);
+            var counter = 0;
+            string line;
+            while ((line = _streamReader.ReadLine()) != null)
+            {
+                Console.WriteLine(line);
+                counter++;
+            }
+            _streamReader?.Close();
+            Console.WriteLine($"File has {counter} lines.");
         }
 
         public void Log(string message)
         {
-            streamWriter.Write(message);
+            _streamWriter = new StreamWriter(PathToFile);
+            _streamWriter.Write(message);
+            _streamWriter?.Close();
         }
 
-   }
-}
\ No newline at end of file
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        protected virtual void Dispose(bool disposeManagedResources)
+        {
+            Console.WriteLine("protected virtual void Dispose");
+            if (_disposed) return;
+            if (disposeManagedResources)
+            {
+                _streamWriter?.Dispose();
```

Comment:
there is no sense to _streamWriter/_streamReader as class-members. and no sense to have "Dispose" method. better to use "using" in specific method

Reply:
in this case we can skip IDispose implementation, but in the second PR I would like to implement it because it's a part of the home task